### DISCUSSION TITLE
Expedition 13 crash

### DISCUSSION THREAD

**Discussion by Terkup** [leodium/leodium-ReLive-NoMansSky-Expeditions-Offline#6] [General]
In expedition 13 the game crashes to desktop upon claiming the phase 4 reward (The Ship of the Damned frigate) although the ship did pop up in the quicksilver vendor after when I checked on my main save. 
Might be a new issue, given that Worlds Part 1 had literally just released when I got to this point.
Attached a screengrab of the error if it helps.
![image](https://github.com/user-attachments/assets/54631227-8941-4ed3-82fd-913e02bb461a)

Sorry about writing my bug report in this here! couldn't find the issues button (which I assume is a choice on your part) but I never really use github so I dunno.

**Comment by cwmonkey** [comment]
@Terkup Have you claimed the The Ship of the Damned frigate on another save? Just wondering if claiming the frigate crashes the game, or if it's just the reward from the expedition.

**Reply by Terkup** [reply]
Nope, never claimed it before, mikssed Adrift when it was on originally.
Also gonna add here, been doing expedition 12, in expedition 12 the "Unbounded" milestone doesn't have any rewards, stopping progress as it needs to give the player the blueprint for "Memory of Bones" for the subsequent memory milestone

**Reply by cwmonkey** [reply]
You can grab my version of 12 here: https://cwmonkey.github.io/nms-expeditions/

It replaces that reward so you can progress. FYI @leodium ^

**Reply by cwmonkey** [reply]
Though @Terkup what I was wondering is if just claiming the frigate outside of the expedition would crash your machine. Could you give that a try just to give us another data point?

**Reply by cwmonkey** [reply]
Oh, another question, did you have a freighter when you claimed the frigate? I remember there being an issue with not having a freighter at that point and claiming the reward, but I forget exactly what it was.

**Reply by leodium** [reply]
> You can grab my version of 12 here: https://cwmonkey.github.io/nms-expeditions/
> 
> It replaces that reward so you can progress. FYI @leodium ^

It was already done! 
I've just made a small change today to make sure ... but it's your json that's been there all along ... It's only this small change in the second line that's been modified.

**Reply by Komander-3** [reply]
> Oh, another question, did you have a freighter when you claimed the frigate? I remember there being an issue with not having a freighter at that point and claiming the reward, but I forget exactly what it was.

I have the same issue as @Terkup,  so I feel I could also give a reply
I sure did own a frigate. I did *not*  try checking the rewards after, I still hope to finish the expedition 🤪... but I'm currently running through Adrift on the actual Adrift Update aswell (downgraded the game) to see if it could be caused by the Worlds update itself

**Reply by Terkup** [reply]
> Though @Terkup what I was wondering is if just claiming the frigate outside of the expedition would crash your machine. Could you give that a try just to give us another data point?

Oh sorry I must've understood you, freighter vs frigate and all that. I am honestly not sure if I had one at the time, and I don't have the save any more. Was there a freighter reward in a milestone or something? If so I did have one for sure, had every other milestone completed and claimed. If not I am not even sure how you'd get one in that expedition given that there's no npcs

**Reply by cwmonkey** [reply]
> > You can grab my version of 12 here: https://cwmonkey.github.io/nms-expeditions/
> > It replaces that reward so you can progress. FYI @leodium ^
> 
> It was already done! I've just made a small change today to make sure ... but it's your json that's been there all along ... It's only this small change in the second line that's been modified.

@leodium I still see the old version: https://github.com/leodium/leodium-ReLive-NoMansSky-Expeditions-Offline/blob/main/Expedition%2012%20-%20Omega/cache/SEASON_DATA_CACHE.JSON

When I diff the patched version from https://cwmonkey.github.io/nms-expeditions/ I get the following:
![expedition-12-diffs](https://github.com/user-attachments/assets/c1550bfa-e1ea-4fba-b5b7-28eb8e71a0cb)

I've attached the fully patched json: 
[SEASON_DATA_CACHE-expedition-12-cwm.JSON](https://github.com/user-attachments/files/16322099/SEASON_DATA_CACHE-cwm.JSON)

**Reply by cwmonkey** [reply]
As for your issue @Komander-3 and @Terkup, as long as you can get the reward from the QS vendor and you are able to complete the expedition, I think we may keep it as-is and just add a note to this expedition that claiming the reward for Phase 4 crashes the game.

At least, until I can think of a workaround. Since this is not a mod our options are limited to swapping out milestones, "batches" of rewards and changing a few specific settings. In this case I don't want to change the reward because it would then not unlock on your main account.

I'll update this thread if I come up with something.

**Comment by Komander-3** [comment]
I got this crash when playing on the steamdeck. I wonder if proton could be the cause of this or if it happens on all systems overall?

**Reply by Terkup** [reply]
It's not proton, I am playing on my desktop computer

**Comment by leodium** [comment]
A chaque nouvelle mise à jour, il y a des changements dans la base de
données. Donc 🤞🤞

**Reply by Komander-3** [reply]
So i completed Adrift on downgraded version and it worked just fine

...which, in hindsight, is not that surprising 😂

**Comment by leodium** [comment]
Comment fais-tu pour avoir une version de jeu rétrogradée ? Un lien ou un
tuto ? 🤞

**Comment by floodie24** [comment]
so i had the same problem, here is how i solved it.. i claimed all rewards except the phase 4 that made me crash. I docked in the anomaly ( no idea if this matters) and exit ship to create a save point, next i claimed reward and crash. now i remove offline mode, reenter the game and load up the adrift save. expedition over, but i can go to quicksilver store and claim my frigate rewards.

**Reply by cwmonkey** [reply]
Yes, in case it wasn't clear in the comments above: the solution is simply don't claim the reward and end the expedition after you've claimed everything else.

FWIW I added a note on my site with the following verbiage:

> The Phase 4 reward (the Ship of the Damned freighter) is broken. Claiming this reward will crash your game. **Unlocking this reward will still unlock the Ship of the Damned freighter in the Quicksilver vendor,** you just can't claim it on the expedition. To end the expedition you must replace the SEASON_DATA_CACHE.JSON with one from a different season, or restart the game in online mode.

**Reply by milchmann88** [reply]
This doesn't work as a workaround when you have other milestones you still need to complete that needs to stay active while doing it.

For example there's the one where you have to stare at the stars for 0/60 seconds.. Problem is, you need to have that milestone as your active quest, BUT the games keeps switching to "claim your phase 4 reward" and therefore you are unable to complete any milestone that needs to be active while doing it.
This prevents me from completing the expedition and claiming the Iron Vulture because the game crashes when trying to claim phase 4.

**Reply by cwmonkey** [reply]
@milchmann88 Ah, that sucks. You can get around this by editing the `SEASON_DATA_CACHE.JSON`. Open the file, search for `60,` and change it to something like `1,` so it completes right away. `0,` would make the quest auto complete. You'll need to make sure the game isn't running when you make your modifications.

**Reply by milchmann88** [reply]
Thanks for the quick reply! I actually worked around it by changing "IsOptional" to true and "IsStageControl" to false under the Phase 4 section and changed it back after doing the rest of the milestones :) That seemed to prevent the game from nagging me into turning it in.

**Reply by cwmonkey** [reply]
Ah, that's interesting. I don't know what `IsStageControl` is but `IsOptional` is exactly what it sounds like (allows you to skip the milestone and still complete the phase/expedition). I'll test setting that by itself and see if it clears the nag. Wouldn't fix the overall problem, but it would make it less annoying.

**Reply by cwmonkey** [reply]
@milchmann88 Are you sure this prevented you from completing the 60 second milestone? I just tested it on PC with the default settings and it worked fine. It would cycle to the phase 4 complete message but that never stopped the countdown.

**Comment by milchmann88** [comment]
Is there any way to flag the phase 4 as completed in your save /expedition data so I can complete the expedition? As stated above, this bug prevents me from completing phase for and therefore claiming the main reward.

**Reply by cwmonkey** [reply]
(workaround in https://github.com/leodium/leodium-ReLive-NoMansSky-Expeditions-Offline/discussions/6#discussioncomment-10268505)

**Comment by leodium** [comment]
But ... Erm... I'm on holiday. Away from my keyboard and these expeditions
🌊

Thanks to you for looking into how to improve the files!!!!

.Boris #afk 🤐

Le jeu. 8 août 2024 à 08:18, Gerald Burns ***@***.***> a
écrit :

**Comment by Sannemen** [comment]
I have roughly the same problem, crashing when claiming the reward for milestone 4 (the "Ship of the Damned" Cursed Frigate).

I've completed the entire expedition, including claiming milestones 1, 2, 3, and 5, and have claimed the final reward (the "Iron Vulture" ship), and I've actually checked, even while offline those already appear in my (regular save) quicksilver vendor's "collect rewards" screen.

So at the stage all the work is done, and the rewards count as "collected", both for this save and for the others.

But, without completing milestone 4, there's no way to convert the save from Expedition to Normal. :(

Is there a way to edit this save to mark that milestone as completed, so the save can be converted back to a regular one?

**Reply by Sannemen** [reply]
Managed to get it edited (and thanks to @rosareven for helping me figure it out!), here's the steps:

- assuming you're starting from everything else collected, except the final reward for milestone 4
- quit the game, backup your save for safety
- load your save slot in [goatfungus' NMS Save Editor](https://github.com/goatfungus/NMSSaveEditor)
- On the Edit menu, pick Edit Raw JSON
- browse to `CommonStateData` > `SeasonState` > `RewardsCollected`. At this state, it should read:
  ```json
  [
    511,
    255,
    255,
    255,
    511
  ]
  ```
- edit the 4th number, replace `255` with `511`, so that it reads
  ```json
  [
    511,
    255,
    255,
    511,
    511
  ]
  ```
- close the window, accept saving the changes
- close the editor, open the game

From that point on, when I went to load up the game, it was tagged as "Normal" (as opposed to "Community Expedition"). I loaded in (I was on foot on the anomaly) and when I tried to start walking I immediately got the "you're done, your save is now converted to a regular one" dialog, along with actual NPCs spawning in the station.

**Reply by cwmonkey** [reply]
I bet just changing the quest to optional like this comment mentioned would have worked:

https://github.com/leodium/leodium-ReLive-NoMansSky-Expeditions-Offline/discussions/6#discussioncomment-10269607

I'll take a look at that as well.

**Reply by cwmonkey** [reply]
Oh, I just remembered, all you had to do is go online and restart the game to end the expedition and convert your save to normal. This was mentioned elsewhere.

I also just tested making Phase 4 optional and that works as well, so I'll update the json file.

**Comment by cwmonkey** [comment]
Testing this now:

1. Having Phase 4 completed but not claimed does not seem to interrupt the milestone to stare at the sky for 60 seconds
2. If you claim the frigate BEFORE you claim the freighter, the phase will remain completed and the game won't crash, but you won't get the frigate. But at least Phase 4 won't nag you.
3. Setting `"IsOptional": true,` AND `"IsStageControl": false,` on Phase 4 breaks the expedition interface for Phase 4 pretty good
4. Settjung JUST `"IsOptional": true,` on Phase 4 allows you to end the expedition with a warning that there is an optional milestone uncompleted: ![image](https://github.com/user-attachments/assets/9c58d2d7-055a-4397-81b4-19caca05c13a)
5. Simply going online in Steam then restarting the game will end the expedition


I'll update the json file to make Phase 4 optional so players can more easily end the expedition

**Reply by cwmonkey** [reply]
You can grab the patched json here: https://cwmonkey.github.io/nms-expeditions/

or from this repo as of [this commit](https://github.com/leodium/leodium-ReLive-NoMansSky-Expeditions-Offline/commit/4e9c3cbca95d4bf14ed0c347af3ba2447cffc7d7)

**Comment by cwmonkey** [comment]
Looks like HG has fixed this. Just tested with the latest cache json in this and my repo and it works with no crashing.
